src/detectors/BTOF/BTOFHitDigi.h

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Souvik Paul, Chun Yuen Tsang, Prithwish Tribedy
+// Special Acknowledgement: Kolja Kauder
+
+// A general digitization for CalorimeterHit from simulation
+// 1. Smear energy deposit with a/sqrt(E/GeV) + b + c/E or a/sqrt(E/GeV) (relative value)
+// 2. Digitize the energy with dynamic ADC range and add pedestal (mean +- sigma)
+// 3. Time conversion with smearing resolution (absolute value)
+// 4. Signal is summed if the SumFields are provided
+//
+// Author: Souvik Paul, Chun Yuen Tsang
+// Date: 19/07/2024
+
+#pragma once
+
+#include "TF1.h"
+#include <iostream>
+#include <memory>
+#include <random>
+#include <vector>
+
+#include <DD4hep/Detector.h>
+#include <algorithms/algorithm.h>
+#include <edm4eic/RawTrackerHitCollection.h>
+#include <edm4hep/SimTrackerHitCollection.h>
+#include <spdlog/spdlog.h>
+
+#include "DD4hep/Detector.h"
+#include "DDRec/Surface.h"
+#include <DDRec/CellIDPositionConverter.h>
+
+#include "BarrelTOFNeighborFinder.h"
+#include "algorithms/digi/BTOFHitDigiConfig.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+class BTOFHitDigi : public WithPodConfig<BTOFHitDigiConfig> {
```

Comment:
Please can you convert this to use the algorithms interface we're now using for our other algorithms. e.g. https://github.com/eic/EICrecon/blob/main/src/algorithms/digi/SiliconTrackerDigi.h

This will require matching changes to BTOFHitDigi.cc `process` function and BTOFHitDigi_factory.h to match.

Reply:
Must process be const? Is that required for multi-threading? It's mostly a concern when I want to use TF1::SetParameter in process for pulse generation. I can get around by declaring new TF1 on stack in process, but I don't know if that's a efficient way to do it.

Reply:
I removed the TF1 entirely. Now I just call the lambda that is used to construct the TF1 directly in TOFPulseGeneration. Leaving it as const should be fine.